common/src/main/java/earth/terrarium/hermes/api/defaults/FillAndBorderElement.java

```diff
@@ -0,0 +1,81 @@
+package earth.terrarium.hermes.api.defaults;
+
+import com.teamresourceful.resourcefullib.common.color.Color;
+import com.teamresourceful.resourcefullib.common.color.ConstantColors;
+import earth.terrarium.hermes.api.TagElement;
+import earth.terrarium.hermes.utils.ElementParsingUtils;
+import net.minecraft.client.gui.GuiGraphics;
+
+import java.util.Map;
+import java.util.function.IntFunction;
+
+public abstract class FillAndBorderElement implements TagElement {
+
+    /* For background fills and borders.
+    Though terms are borrowed from CSS's Box Model, does not change Hermes' layout model.
+    Does allow inheriting elements to be set taller and/or wider.
+    The area behind the element, plus any padding, may be filled with background color
+    A border may surround the padding area, with its own width and color. */
+
+    protected int backgroundPadding = 0;
+    protected Color backgroundColor = ConstantColors.gray;
+    protected boolean hasBackground = false;
+    protected int borderWidth = 0;
+    protected Color borderColor = ConstantColors.whitesmoke;
+    protected boolean hasBorder = false;
+    protected int hSpacing; // (background + borderWidth);
+    protected int vSpacing; // 1 or (backgroundPadding + borderWidth)
+
+    protected FillAndBorderElement(Map<String, String> parameters) {
+        if (parameters.containsKey("background")) {
+            this.hasBackground = true;
+            String[] backgroundSpecs = parameters.get("background").split(" ");
+            switch (backgroundSpecs.length) {
+                case 2:
+                    this.backgroundColor = ElementParsingUtils.tryParse(backgroundSpecs[1], Color::parse, backgroundColor);
+                case 1:
+                    this.backgroundPadding = ElementParsingUtils.tryParse(backgroundSpecs[0], Integer::parseInt, backgroundPadding);
+            }
+        }
+        if (parameters.containsKey("border")) {
+            this.hasBorder = true;
+            this.borderWidth = 1; // Set a default value when the tag has a border attribute.
+            String[] borderSpecs = parameters.get("border").split(" ");
```

Comment:
both border and color do this so it should become an element parsing util and have its own type that it returns

Reply:
also when this is done to its own type you can remove the hasX booleans and just check if type is null

Reply:
Sorry, not sure I fully follow.

I went with special parsing for these attributes, like `template` for `<columns>`, because my intent was to combine two values in each of these tags, for brevity, and also provide four possibilities to the user: 

(Example with just `border`, but `background` is similar)

- no `border` attribute supplied -> _do nothing_
- `border=""` supplied -> draw a border with _default_ thickness `1` and default `whitesmoke` color.
- `border="x"` supplied -> draw border with `x` thickness and default color
- `border="x y"` supplied -> draw border with `x` thickness and `y` color

So, as I understand it, there are two cases a `null` would conflate: no attribute specified, and no value supplied by the user to over-ride coded defaults.

---
\*Granted, the 2nd case, currently would probably be only used for aiding the description writer when tweaking fancy layouts; this is also helpful for some entities that render quite a bit outside their bounding box, such as the squid. For an `<entity>`, it's background fill shows Hermes’ “understanding” of where the entity is, and the writer can use the entity tag’s special attributes adjust for mobs that need it.

There is also the possibility to have the defaults be themed, but I’m not interested in that now, tbh, and would rather leave it out of scope for this PR.

Reply:
For this I am talking about the value pair of an int and color so you can call it IntColorPair and have it parse on split and have defaults for the color passed in but if you ever need to add more this constructor will become massive so doing it prematurely is needed.

Reply:
Got it; working on it.

Reply:
b6c6582 does it.